lib/fetchers/url.rb

```diff
@@ -178,5 +163,42 @@ def download_archive(path)
       @temp_archive_path = nil
       final_path
     end
+
+    def http_opts
+      opts = {}
+      opts[:ssl_verify_mode] = OpenSSL::SSL::VERIFY_NONE if @insecure
+
+      if @config['server_type'] == 'automate'
```

Comment:
I think we should deprecate all delivery specific handling in this class and move all the required logic into audit cookbook `files/default/vendor/chef-automate/fetcher.rb`. I understand that we do not want to make this in one move. But I'd like to see a follow-up on that PR to cover that. This allows us to have the chef-* specific parts in one place.